integration/simulation/validate_chain.go

```diff
@@ -145,7 +145,7 @@ func checkBlockchainOfObscuroNode(t *testing.T, node *host.Node, minObscuroHeigh
 	l1Height := node.DB().GetCurrentBlockHead().Height
 
 	// check that the L1 view is consistent with the L1 network.
-	if maxEthereumHeight-l1Height > MAX_BLOCK_DELAY {
+	if int64(maxEthereumHeight)-int64(l1Height) > MAX_BLOCK_DELAY {
 		t.Errorf("Obscuro node %d fell behind %d blocks.", obscurocommon.ShortAddress(node.ID), maxEthereumHeight-l1Height)
```

Comment:
Does this need to be casted ?

Reply:
Yeah. Otherwise, it overflows with a massive "delay" if `l1Height` exceeds `maxEthereumHeight` (I saw this happening in the tests).

Reply:
> Yeah. Otherwise, it overflows with a massive "delay" if `l1Height` exceeds `maxEthereumHeight` (I saw this happening in the tests).

umm, but how could it exceed it? Since the maxEthereumHeight is calculated on the actual L1 nodes. 
Can you raise a ticket to investigate this? My fear is that it could hide some problem

Reply:
I'm guessing what happens is that we calculate the ethMaxHeight, but the sim keeps running and adds more to certain L1 chains. As we discussed yesterday, even before when we were stopping the sim before validating, both processes happened in parallel (because we don't wait for the stop to complete). We could perhaps wait for all the nodes to stop before performing the validation?

For now, I'll add a todo and raise an issue.

Reply:
ah, yes. that's the reason. Can you add that in the comment there? No need to raise an issue.

Reply:
Sure.